File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxPro = 0
        minPrice =prices[0]
        for i in range(len(prices)):
            minPrice = min(minPrice, prices[i])
            profit=prices[i] - minPrice
            maxPro = max(maxPro,profit)
        return maxPro     

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxprofit=0;
    let minprice=prices[0];
    for(let num of prices){
        if(num<minprice){
            minprice=num;
        }
        
        let profit=num-minprice;
        if(profit>maxprofit){
            maxprofit=profit;
        }
        
    }
    return maxprofit;
};